Updates to Considerations document (#388)
Microsoft/api-guidelines: azure/ConsiderationsForServiceDesign.md

:ballot_box_with_check: **YOU SHOULD**  capture what you have learned during the preview stage and share these findings with your team and with the API Stewardship Board.

> ## Communicate Deprecations

Reviewer: @JeffreyRichter @markweitzel Should we introduce guidance around scream testing so that teams are allowed to start returning different status code after the deadline has passed before effectively pulling the API version support?

Reviewer 2: If the service is still running with the latest version, then the client's request will reach the endpoint, the service will examine the passed-in api-version query parameter, determine that the version is no longer valid and return a 400? with an Azure error response code of "Invalid api-version"?

Mark, we should standardize the status code and Azure error code/message when a client makes a request to an invalid api-version and make service team return this consistently.

> As your service evolves over time, it will be natural that you want to remove operations that are no longer needed. For example, additioanl requirements or new capability in your service, may have resulted in a new operation that, effectively, replaces an old one.

Reviewer: Nit
```suggestion
As your service evolves over time, it will be natural that you want to remove operations that are no longer needed. For example, additional requirements or new capability in your service, may have resulted in a new operation that, effectively, replaces an old one.
```

> As your service evolves over time, it will be natural that you want to remove operations that are no longer needed. For example, additioanl requirements or new capability in your service, may have resulted in a new operation that, effectively, replaces an old one.

Reviewer 2: ```suggestion
You service should evolve over time in a backward-compatible way. However, occasionally, a service may introduce a breaking change (with Azure Breaking Change Board approval). When a breaking change is approved, old service versions are deprecated and will eventually stop working.
```

Reviewer: Nit - `You service` -> `Your service`

> As your service evolves over time, it will be natural that you want to remove operations that are no longer needed. For example, additional requirements or new capability in your service, may have resulted in a new operation that, effectively, replaces an old one.

Reviewer 3: I wouldn't say it's "natural."  I want people to think about public APIs like marriage - it's forever unless you're so unhappy supporting it you're willing to pay for a very expensive breaking change lawyer.

```suggestion
As your service evolves over time, you may need to remove operations that are no longer supportable. For example, additional requirements or a new capability in your service may result in a new operation that effectively replaces an old one.
```

> Azure has a well established breaking changes policy that describes how to approach these kinds of changes. As part of this policy, the service team is required to clearly communicate to customers when their API is changing, e.g. deprecating operations. Often, this is done via an email to the address that is attached to the Azure subscription.

Reviewer 2: ```suggestion
Azure has a well-established breaking changes policy that describes how to approach these kinds of changes. As part of this policy, the service team is required to clearly communicate to customers when a currently-shipping api-version will stop working. Often, this is done via an email to the address that is attached to the Azure subscription.
```

> However, given how many organizations are structured, it's common that this email address is different from the actual people writing code against your API. To address this, the service API should declare that it may return the `azure-deprecating` header, to indicate that this operation will be removed in the future. There is a simple string convention, specied in the [Azure REST API Guidelines](https://aka.ms/azapi/guidelines) that provides more information about the forthcoming deprecation.

Reviewer: See other header name proposal on https://github.com/microsoft/api-guidelines/pull/387

> However, given how many organizations are structured, it's common that this email address is different from the actual people writing code against your API. To address this, the service API should declare that it may return the `azure-deprecating` header, to indicate that this operation will be removed in the future. There is a simple string convention, specied in the [Azure REST API Guidelines](https://aka.ms/azapi/guidelines) that provides more information about the forthcoming deprecation.

Reviewer 2: ```suggestion
However, given how many customer organizations are structured, it's common that this email address is different from the actual people writing code calling into an Azure service. To address this, the service API should declare that it may return the `azure-deprecating` header, to indicate that this service version will stop working in the future. There is a simple string convention, specied in the [Azure REST API Guidelines](https://aka.ms/azapi/guidelines) that provides more information about the deprecation.
```

Reviewer 3: All of Jeff's suggestions, plus a bit at the end
```suggestion
However, given how many organizations are structured, it's common that this email address is different from the actual people writing code against your API. To address this, the service API should declare that it may return the `azure-deprecating` header, to indicate that this operation will be removed in the future. There is a simple string convention, specified in the [Azure REST API Guidelines](https://aka.ms/azapi/guidelines#putdirectlinkhere?) that provides more information about the forthcoming deprecation and actions your customers should take.
```

> This header is targeted at developers or operation professionals, and it is intended to give them enough information and lead time to properly adapt to this change. Your documentation should reference this header and encourage logging and alerting practices based on its presence.

Reviewer 2: ```suggestion
This header is targeted at developers or operation professionals, and it is intended to give them enough information and lead time to properly adapt to the necessary change. Your documentation should reference this header and encourage logging and alerting practices based on its presence.
```

> This header is targeted at developers or operation professionals, and it is intended to give them enough information and lead time to properly adapt to this change. Your documentation should reference this header and encourage logging and alerting practices based on its presence.

Reviewer 3: It might be worth filing a tracking bug to come update this in 6 months with a description of what the SDKs do with the header for additional context?
